### PR TITLE
Analyzer for disallowing Json.NET TypeNameHandling values other than None

### PR DESCRIPTION
**Pull request by dotpaul**

### CODE REVIEW

**Comment by dotpaul** [comment]
FYI @DSIE63 @matt-desai @timhermann

File: src/Microsoft.NetCore.Analyzers/Core/Security/JsonNetTypeNameHandling.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                        compilationStartAnalysisContext.Compilation);
+
+                    if (!wellKnownTypeProvider.TryGetTypeByMetadataName(
+                            WellKnownTypeNames.NewtonsoftJsonTypeNameHandling,
+                            out INamedTypeSymbol typeNameHandlingSymbol))
+                    {
+                        return;
+                    }
+
+                    bool IsOtherThanNone(IFieldReferenceOperation fieldReferenceOperation)
```
Comment (mavasani) [inline src/Microsoft.NetCore.Analyzers/Core/Security/JsonNetTypeNameHandling.cs]: Nit: Move the local functions at the end of the method, and add an explicit return statement prior to local function.

**Comment by dotpaul** [comment]
Ping @LLLXXXCCC

**Comment by dotpaul** [comment]
Thank you @mavasani and @LLLXXXCCC !

**Pull request merged**